packages/asset-swapper/src/utils/market_utils.ts

```diff
@@ -0,0 +1,89 @@
+import { schemas } from '@0x/json-schemas';
+import { BigNumber } from '@0x/utils';
+import * as _ from 'lodash';
+
+import { constants } from '../constants';
+import {
+    MarketOperation,
+    PrunedSignedOrder,
+} from '../types';
+
+import { assert } from './assert';
+
+export const marketUtils = {
+    findOrdersThatCoverTakerAssetSellAmount(
+        orders: PrunedSignedOrder[],
+        takerAssetSellAmount: BigNumber,
+        slippageBufferAmount: BigNumber = new BigNumber(0),
+    ): { resultOrders: PrunedSignedOrder[]; remainingFillAmount: BigNumber } {
+        return findOrdersThatCoverAssetFillAmount(
+            orders,
+            takerAssetSellAmount,
+            MarketOperation.Sell,
+            slippageBufferAmount,
+        );
+    },
+    /**
+     * Takes an array of orders and returns a subset of those orders that has enough makerAssetAmount
+     * in order to fill the input makerAssetFillAmount plus slippageBufferAmount. Iterates from first order to last order.
+     * Sort the input by ascending rate in order to get the subset of orders that will cost the least ETH.
+     * @param   orders                      An array of objects that extend the Order interface. All orders should specify the same makerAsset.
+     *                                      All orders should specify WETH as the takerAsset.
+     * @param   makerAssetFillAmount        The amount of makerAsset desired to be filled.
+     * @param   opts                        Optional arguments this function accepts.
+     * @return  Resulting orders and remaining fill amount that could not be covered by the input.
+     */
+    findOrdersThatCoverMakerAssetBuyAmount(
+        orders: PrunedSignedOrder[],
+        makerAssetBuyAmount: BigNumber,
+        slippageBufferAmount: BigNumber = new BigNumber(0),
+    ): { resultOrders: PrunedSignedOrder[]; remainingFillAmount: BigNumber } {
+        return findOrdersThatCoverAssetFillAmount(
+            orders,
+            makerAssetBuyAmount,
+            MarketOperation.Buy,
+            slippageBufferAmount,
+        );
+    },
+};
+
+function findOrdersThatCoverAssetFillAmount(
+    orders: PrunedSignedOrder[],
```

Comment:
I think these are sorted right? Should we just name the variable `sortedOrders` so its clear some pre-work is required.